src/base/bittorrent/torrentimpl.cpp

```diff
@@ -2120,6 +2123,9 @@ void TorrentImpl::updateStatus(const lt::torrent_status &nativeStatus)
         else if (isDownloading())
             m_unchecked = true;
     }
+
+    while (!m_statusUpdatedTriggers.isEmpty())
+        std::invoke(m_statusUpdatedTriggers.dequeue());
```

Comment:
I would just invoke it directly: `m_statusUpdatedTriggers.dequeue()();` and save the unusual `std::invoke` for metaprogramming.

Reply:
> I would just invoke it directly: `m_statusUpdatedTriggers.dequeue()();` and save the unusual `std::invoke` for metaprogramming.

`std::invoke()` looks more clear than `m_statusUpdatedTriggers.dequeue()();`, IMO. At least, at a cursory glance, `std::invoke()` immediately indicates that some function is being called here, while `()()` is usually lost.

Reply:
And even more so, I don't have the slightest feeling that `std::invoke()` is intended only for metaprogramming. It makes any call that differs in spelling from the ordinary one clearer. For example, `std::invoke([]() { do_something(); });` looks much more readable than `([]() { do_something(); })();`.

Of course, you can say "just copy it to another variable, and then use an ordinary call to make it look readable." But is it really worth it?